In [57]:
import numpy as np
import pandas as pd
import math
from math import factorial
import warnings
warnings.filterwarnings('ignore')

#### Задание-1

In [58]:
# Известно, что генеральная совокупность распределена нормально
# со средним квадратическим отклонением, равным 16.
# Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
# если выборочная средняя M = 80, а объем выборки n = 256.

In [59]:
# Решение: так как сигма нам известна, то используем Z-таблицу и применяем формулу
# X(+/-)Z(a/2)*sig/sqrt(n)
# Z(97.5%)=1.96

In [60]:
M = 80
X = M
n = 256
sig = 16
Z = 1.96
int1 = X - Z*(sig / np.sqrt(n))
int1

78.04

In [61]:
int2 = X + Z*(sig / np.sqrt(n))
int2

81.96

*Ответ: доверительный интервал будет расположен в границах [78.04; 81.96].*

#### Задание-2

In [62]:
# В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
# получены опытные данные:
# 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
# Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
# оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
# значение с доверительной вероятностью 0,95.

In [63]:
# Решение: так как сигма нам не известна, то используем t-таблицу, 
# а учитывая, что выборка состоит из 10 измерений - для рассчета сигмы 
# применяем формулу для несмещенного значения.
# Интервал находим по формуле X(+/-)t(a/2)*sig/sqrt(n)

In [64]:
a = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
a

array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [65]:
x = pd.DataFrame(a)
x.mean()

0    6.59
dtype: float64

In [66]:
sig = x.std(ddof=1)
sig

0    0.450802
dtype: float64

In [67]:
# t-коэффициент для выборки из десяти элементов равен 2.2621
t = 2.2621

In [68]:
x1 = x.mean() + t * sig / np.sqrt(n)
x1

0    6.653735
dtype: float64

In [69]:
x2 = x.mean() - t * sig / np.sqrt(n)
x2

0    6.526265
dtype: float64

*Ответ: истинное значение величины Х находится в интервале [6.526265; 6.653735] с вероятностью 95%.*

#### Задание-3

In [70]:
# Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
# Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
# оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [71]:
# Решение:
# Н0: диаметр = 17 мм, mu=mu0
# Н1: диаметр = 17.5 мм, mu>muo
# Критерий Z-тест (сигма известна) и для α=0,05 ZH0 = 1.645
# Найдем ZH1 и сравним его с ZH0. если гипотеза о том, чсто средний диаметр шариков = 17 мм верна,
# то ZH0 будет больше ZH1.

In [72]:
D = 4
n = 100
sig = np.sqrt(D)
ZH1 = (17.5 - 17) / (sig / np.sqrt(n))
ZH1

2.5

*Ответ: нулевая гипотеза неверна, шарики для подшипников, 
изготовленные автоматическим станком, имеют средний диаметр не 17 мм.*

#### задание-4

In [73]:
# Продавец утверждает, что средний вес пачки печенья составляет 200 г. 
# Из партии извлечена выборка из 10 пачек.
# Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. 
# Известно, что их веса распределены нормально. 
# Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [74]:
# Решение: введем нулевую гипотезу - Н0: mu0=200
# H1: mu != mu0.
# Так как сигма неизвестна применим t-критерий Стьюдента.
mu0 = 200
a = 0.01
n = 10

In [75]:
a = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
a

array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [76]:
# Найдем среднее значение веса пачки:
x = pd.DataFrame(a)
x.mean()

0    198.5
dtype: float64

In [77]:
# Вычислим несмещенное среднеквадратичное отклонение:
sig = x.std(ddof=1)
sig

0    4.453463
dtype: float64

In [78]:
# По таблице критических точек распределения Стьюдента найдем критическую точку
# по уровню значимости α/2 = 0,005. 
t = 3.6896

In [79]:
#  Вычислим наблюдаемое t по формуле:
t_n = (x.mean() - mu0) / sig * np.sqrt(n)
t_n

0   -1.065107
dtype: float64

*Ответ: Так как t_n наблюдаемое меньше t (-1.065107 < 3.6896), то нулевую гипотезу о равенстве среднего веса 200 грамм можно принять.*